torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -931,3 +931,156 @@ def call_triton_kernel(self, variable, args, kwargs, tx):
+        >>> #             'in_ptr0': x_1, 'in_ptr1': y_1, 'out_ptr': empty_like,
+        >>> #             'n_elements': 3, 'BLOCK_SIZE': 16
+        >>> #         })
+        >>> #     return empty_like
+
+    """
+    from triton.runtime.autotuner import Autotuner
+    from triton.runtime.jit import JITFunction
+
+    if not isinstance(triton_kernel, (JITFunction, Autotuner)):
+        raise RuntimeError(
+            "capture_triton only works on functions annotated with triton.jit or triton.autotune"
+        )
+    return TraceableTritonKernelWrapper(triton_kernel, None, None)
+
+
+from ..fx._symbolic_trace import is_fx_tracing
+
+
+class TracingTritonHOPifier(TritonHOPifier):
+    def raise_unsupported(self, msg):
+        raise RuntimeError(msg)
+
+    def is_callable(self, maybe_callable):
+        return callable(maybe_callable)
+
+    def call_grid(self, grid, meta, tx):
+        assert tx is None
+        return grid(meta)
+
+    def check_grid(self, grid):
+        if not isinstance(grid, tuple):
+            raise RuntimeError(
+                "capture_triton can only handle grids that resolve to Tuple[int, ...]."
+            )
+        return grid
+
+    def call_HOP(self, variable, grids, combined_args, tx):
+        assert tx is None
+        import torch
+
+        non_graphable_args = {
+            k: v
+            for k, v in combined_args.items()
+            if not isinstance(v, (torch.Tensor, int, float, bool))
+        }
+        graphable_args = {
+            k: v
+            for k, v in combined_args.items()
+            if isinstance(v, (torch.Tensor, int, float, bool))
+        }
+        meta = graphable_args
+
+        constant_args_idx = kernel_side_table.add_constant_args(non_graphable_args)
+        return triton_kernel_wrapper_mutation(
+            kernel_idx=variable.kernel_idx,
+            constant_args_idx=constant_args_idx,
+            grid=grids,
+            kwargs=meta,
+        )
+
+
+tracing_triton_hopifier_singleton = TracingTritonHOPifier()
+
+
+class TraceableTritonKernelWrapper:
+    def __init__(self, kernel, kernel_idx, grid):
+        self.kernel = None
+        self.grid = None
+        tracing_triton_hopifier_singleton.init_variable(self, kernel, kernel_idx, grid)
+        assert self.kernel is not None
+
+    def __getitem__(self, *args):
+        return tracing_triton_hopifier_singleton.call_getitem(self, args)
+
+    def run(self, *args, **kwargs):
+        import torch._dynamo
+
+        if not is_fx_tracing() or torch._dynamo.is_compiling():
+            assert self.kernel is not None
+            return self.kernel.run(*args, **kwargs)
+        return tracing_triton_hopifier_singleton.call_run(self, args, kwargs, None)
+
+    def __call__(self, *args, **kwargs):
+        import torch._dynamo
+
+        if not is_fx_tracing() or torch._dynamo.is_compiling():
+            assert self.kernel is not None
+            return self.kernel[self.grid](*args, **kwargs)
```

Comment:
be consistent with above 

```suggestion
            return self.kernel.run(*args, **kwargs, grid=self.grid)
```

Reply:
Getting `TypeError: JITFunction.run() missing 1 required keyword-only argument: 'warmup'` when I use `.run` here -- doing the `__call__` doesn't seem to require the warmup arg

Reply:
Pass warmup=False, `__call__` does that: https://github.com/triton-lang/triton/blob/c14b033cd979d5c39e5fdb3847c022fa5d71a0c1/python/triton/runtime/jit.py#L326C59-L326C72

Reply:
Thank you, that worked